include/envoy/event/dispatcher.h

```diff
@@ -19,6 +19,7 @@
 #include "envoy/network/listen_socket.h"
 #include "envoy/network/listener.h"
 #include "envoy/network/transport_socket.h"
+#include "envoy/server/overload/thread_local_overload_state.h"
```

Comment:
Should you consider forward declaration for Server::ThreadLocalOverloadState ?

Reply:
I am aware of the guidelines at: https://google.github.io/styleguide/cppguide.html#Forward_Declarations .

See "The decision is: Try to avoid forward declarations of entities defined in another project."

The classes involved here are both part of the Envoy project so some of the justifications listed don't fully apply in this case.  I would personally use forward declaration to resolve these circular dependencies.  If you prefer to refactor headers, please do it in a separate PR in order to simplify review of functional changes in this PR.